src/main/java/no/fdk/referencedata/filetype/eu/EUFileTypeHarvester.java

```diff
@@ -0,0 +1,113 @@
+package no.fdk.referencedata.filetype.eu;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import no.fdk.referencedata.filetype.FileType;
+import no.fdk.referencedata.filetype.FileTypeHarvester;
+import no.fdk.referencedata.vocabulary.EUNotationType;
+import no.fdk.referencedata.vocabulary.EUVOC;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.vocabulary.DC;
+import org.apache.jena.vocabulary.DCTerms;
+import org.apache.jena.vocabulary.RDF;
+import org.springframework.core.io.UrlResource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import reactor.core.publisher.Flux;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Component
+@Slf4j
+public class EUFileTypeHarvester implements FileTypeHarvester {
+
+    private static final String SEARCH_API = "https://data.europa.eu/api/hub/search";
+
+    public Flux<FileType> harvestFileTypes() {
+        log.info("Starting harvest of EU file types");
+        final org.springframework.core.io.Resource fileTypesRdfSource = getFileTypesSource();
+        if(fileTypesRdfSource == null) {
+            return Flux.error(new Exception("Unable to fetch file-types distribution"));
+        }
+
+        final Model model;
+        try {
+            model = RDFDataMgr.loadModel(fileTypesRdfSource.getURI().toString(), Lang.RDFXML);
+        } catch (IOException e) {
+            log.error("Unable to load model", e);
+            return Flux.error(new Exception("Unable to load model"));
+        }
```

Comment:
Ekstraher ut i egen metode og bruk deretter Mono.justOrEmpty for å håndtere null verdi